NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/AzureAppServiceRoleNameFromHostNameHeaderInitializer.cs

```diff
@@ -72,7 +49,16 @@ public AzureAppServiceRoleNameFromHostNameHeaderInitializer(IHttpContextAccessor
         /// For US Gov Cloud: ".azurewebsites.us"
         /// For Azure Germany: ".azurewebsites.de".
         /// </summary>
-        public string WebAppSuffix { get; set; }
+        public string WebAppSuffix
+        {
+            get => RoleNameContainer.HostNameSuffix;
+
+            set
+            {
+                RoleNameContainer.HostNameSuffix = value;
+                RoleNameContainer.SetFromEnvironmentVariable(out _);
```

Comment:
It looks like `out bool isAzureWebApp` is used as a control leading to `out _` in other place. Is there a way to provide this control outside of `SetFromEnvironmentVariable` method to make operation mode selected separately?

Reply:
I refactored this to initialize from EnvironmentVariable in a constructor.
This required some additional changes, but it makes both use and testing scenarios cleaner. :)